shared-module/displayio/OnDiskBitmap.c

```diff
@@ -115,8 +115,66 @@ void common_hal_displayio_ondiskbitmap_construct(displayio_ondiskbitmap_t *self,
         self->stride = (bit_stride / 8);
     }
 
+    self->dither = false;
+
+    // Dither defaults for 565 display
+    self->dither_mask_r = 7;
+    self->dither_mask_g = 3;
+    self->dither_mask_b = 7;
+}
+
+// Perlin Noise code use to generate dithering from:
+// https://gist.github.com/nowl/828013
+
+float lerp(float a0, float a1, float w) {
+    return (1.0f - w)*a0 + w*a1;
+}
+
+static int SEED = 0;
+
+static int hash[] = {208,34,231,213,32,248,233,56,161,78,24,140,71,48,140,254,245,255,247,247,40,
```

Comment:
By using uint8_t instead of int for the table type, 768 bytes of flash would be saved.  Flash space is at a premium on many circuitpython devices.